common/src/main/java/com/twilio/video/examples/common/VideoFrameKtx.kt

```diff
@@ -0,0 +1,171 @@
+package com.twilio.video.examples.common
+
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.graphics.ImageDecoder
+import android.graphics.ImageFormat
+import android.graphics.Matrix
+import android.graphics.Rect
+import android.graphics.YuvImage
+import android.os.Build
+import tvi.webrtc.VideoFrame
+import tvi.webrtc.YuvConverter
+import java.io.ByteArrayOutputStream
+import java.io.IOException
+import java.nio.ByteBuffer
+
+
+/**
+ * Converts a [tvi.webrtc.VideoFrame] to a Bitmap. This method must be called from a thread with a
+ * valid EGL context when the frame buffer is a [VideoFrame.TextureBuffer].
+ */
+fun VideoFrame.toBitmap() : Bitmap? {
+    return captureBitmap(this)
+}
+
+private fun captureBitmap(frame: VideoFrame) : Bitmap? {
+    return if (frame.buffer is VideoFrame.TextureBuffer) {
+        captureBitmapFromTexture(frame)
+    } else {
+        captureBitmapFromYuvFrame(frame)
+    }
+}
+
+private fun captureBitmapFromYuvFrame(videoFrame: VideoFrame): Bitmap? {
+    val yuvImage = i420ToYuvImage(
+        videoFrame.buffer.toI420(),
+        videoFrame.buffer.width,
+        videoFrame.buffer.height
+    )
+    val stream = ByteArrayOutputStream()
+    val rect =
+        Rect(0, 0, yuvImage.width, yuvImage.height)
+
+    // Compress YuvImage to jpeg
+    yuvImage.compressToJpeg(rect, 100, stream)
+
+    // Convert jpeg to Bitmap
+    val imageBytes = stream.toByteArray()
+    var bitmap: Bitmap?
+    bitmap = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
```

Comment:
We should be able to combine the declaration with the assignment into one statement and remove the type and rely on type inference here